flask_restful/__init__.py

```diff
@@ -272,15 +272,21 @@ def handle_error(self, e):
         """
         got_request_exception.send(current_app._get_current_object(), exception=e)
 
-        if not hasattr(e, 'code') and current_app.propagate_exceptions:
+        if not isinstance(e, HTTPException) and current_app.propagate_exceptions:
             exc_type, exc_value, tb = sys.exc_info()
             if exc_value is e:
                 raise
             else:
                 raise e
 
-        code = getattr(e, 'code', 500)
-        data = getattr(e, 'data', error_data(code))
+        if isinstance(e, HTTPException):
+            code = e.code
+            data = getattr(e, 'data', error_data(code))
```

Comment:
`HTTPException` never has a `data` attribute [AFAIK](https://github.com/mitsuhiko/werkzeug/blob/4bd3fd98f77f1d71dcd26f787abec522a551e2ee/werkzeug/exceptions.py#L83). Lets skip the `getattr` call and use `error_data` every time for instances of `HTTPException`.